build.gradle

```
@@ -404,3 +404,11 @@ def configureGroovydoc(TaskProvider groovydoc) {
     include "spock/**"
   }
 }
+
+tasks.preprocessBranchesAndPrsWorkflow {
+  outputs.file('.github/codecov.yml').withPropertyName('codecovConfig')
+}
+
+tasks.preprocessReleaseWorkflow {
+  outputs.file('.github/codecov.yml').withPropertyName('codecovConfig')
+}
```

🤔 This does not work, iirc Gradle wants a dedicated directory for output caching to work.

https://ge.spockframework.org/s/ybqsxrfvj6d7q/timeline?details=6lhcybvrqmfrk

```
Overlapping outputs: Gradle does not know how file '.github/workflows/docs-pr.yaml' was created (output property 'workflowFile'). Task output caching requires exclusive access to output paths to guarantee correctness (i.e. multiple tasks are not allowed to produce output in the same location).
```

And I think you meant to use `inputs` here.
```suggestion
tasks.preprocessBranchesAndPrsWorkflow {
  inputs.file('.github/codecov.yml').withPropertyName('codecovConfig')
}

tasks.preprocessReleaseWorkflow {
  inputs.file('.github/codecov.yml').withPropertyName('codecovConfig')
}
```

---

Hm, I'm a bit confused now.
The part you quoted complains about `docs-pr.yaml` which does not make sense to me.
The part you linked to complains about `codecov.yml` which does make sense to me because I was stupid.

I did not mean to declare it as input.
Actually it is input **and** output, as the file is read, the number updated and then written again.
Which per-se is sub-optimal, especially for a cacheable task.
And the main problem is, that two different workflows do this and so both have it as output.

I think I'll change it, so that both tasks write to separate different files and then have finalizer tasks that move them over the actual checked in file to decouple the input and output and have no overlapping outputs.

This will only make the task cached after the second run if the codecov file changed, but that should be an acceptable trade-off.

But the question remains where you copied the warning with `docs-pr.yaml` from.